### PR TITLE
MASP SDK Refactor

### PR DESCRIPTION
**Pull request by batconjurer**
Factored most of the masp code out of the sdk and into shielded token crate

## Checklist before merging 
- [ ] If this PR has some consensus breaking changes, I added the corresponding `breaking::` labels
    - This will require 2 reviewers to approve the changes

### CODE REVIEW

File: crates/benches/native_vps.rs [hunk 1]
```diff
@@ -60,6 +56,10 @@ use namada_node::bench_utils::{
     TX_BRIDGE_POOL_WASM, TX_IBC_WASM, TX_INIT_PROPOSAL_WASM, TX_RESIGN_STEWARD,
     TX_TRANSFER_WASM, TX_UPDATE_STEWARD_COMMISSION, TX_VOTE_PROPOSAL_WASM,
 };
+use namada_token::masp::{
```
Comment (tzemanovic) [inline crates/benches/native_vps.rs:59]: let's import this from apps_lib - it re-exports all of sdk
```suggestion
use namada_apps_lib::token::masp::{
```

File: crates/benches/Cargo.toml [hunk 2]
```diff
@@ -43,6 +43,7 @@ namada-eth-bridge = [
 [dev-dependencies]
 namada_apps_lib = { path = "../apps_lib", features = ["benches"] }
 namada_node = { path = "../node", features = ["benches"] }
+namada_token = {path = "../token"}
```
Comment (tzemanovic) [inline crates/benches/Cargo.toml:46]:
```suggestion
```
accessible from apps_lib

File: crates/apps_lib/Cargo.toml [hunk 3]
```diff
@@ -31,10 +32,13 @@ namada-eth-bridge = [
 
 [dependencies]
 namada_core = {path = "../core"}
+namada_io = { path = "../io" }
 namada_macros = {path = "../macros"}
 namada_migrations = {path = "../migrations", optional = true}
 namada_sdk = {path = "../sdk", features = ["download-params", "multicore"]}
+namada_token = {path = "../token"}
```
Comment (tzemanovic) [inline crates/apps_lib/Cargo.toml:39]:
```suggestion
```
token is re-exported from sdk

File: crates/apps_lib/Cargo.toml [hunk 4]
```diff
@@ -23,6 +23,7 @@ benches = ["lazy_static", "namada_sdk/benches"]
 migrations = [
   "namada_migrations",
   "namada_sdk/migrations",
+  "namada_token/migrations",
```
Comment (tzemanovic) [inline crates/apps_lib/Cargo.toml:26]:
```suggestion
```

File: crates/apps_lib/Cargo.toml [hunk 5]
```diff
@@ -31,10 +32,13 @@ namada-eth-bridge = [
 
 [dependencies]
 namada_core = {path = "../core"}
+namada_io = { path = "../io" }
```
Comment (tzemanovic) [inline crates/apps_lib/Cargo.toml:35]: maybe we could also re-export `namada_io` from sdk so we don't have to add this

File: crates/sdk/src/lib.rs [hunk 6]
```diff
@@ -37,33 +37,28 @@ pub mod validation;
...
 use namada_core::address::Address;
 use namada_core::collections::HashSet;
-pub use namada_core::control_flow;
 use namada_core::dec::Dec;
 use namada_core::ethereum_events::EthAddress;
 use namada_core::ibc::core::host::types::identifiers::{ChannelId, PortId};
 use namada_core::key::*;
 use namada_core::masp::{ExtendedSpendingKey, PaymentAddress, TransferSource};
+pub use namada_core::{control_flow, task_env};
+use namada_io::{Client, Io, NamadaIo};
```
Comment (tzemanovic) [inline crates/sdk/src/lib.rs:56]:
```suggestion
use namada_io::{Client, Io, NamadaIo};
pub use namada_io as io;
```
(this might need fmt)

Comment (tzemanovic) [reply crates/sdk/src/lib.rs:56]: I think this should allow to undo lots of changes that use io from sdk

File: crates/node/Cargo.toml [hunk 7]
```diff
@@ -43,10 +43,12 @@ namada-eth-bridge = [
 
 [dependencies]
 namada_apps_lib = {path = "../apps_lib"}
+namada_io = {path = "../io"}
```
Comment (tzemanovic) [inline crates/node/Cargo.toml:46]:
```suggestion
```
should be accessible from same path as before

File: crates/node/Cargo.toml [hunk 8]
```diff
@@ -43,10 +43,12 @@ namada-eth-bridge = [
...
 namada_migrations = {path = "../migrations", optional = true}
 namada_replay_protection = {path = "../replay_protection"}
 namada_sdk = {path = "../sdk", features = ["wasm-runtime"]}
 namada_test_utils = {path = "../test_utils", optional = true}
+namada_token = {path = "../token"}
```
Comment (tzemanovic) [inline crates/node/Cargo.toml:51]:
```suggestion
```
also from `apps_lib::token`

File: crates/tests/Cargo.toml [hunk 9]
```diff
@@ -28,8 +28,9 @@ namada-eth-bridge = [
 
 [dependencies]
 namada_core = {path = "../core", features = ["testing"]}
-namada_sdk = {path = "../sdk", features = ["download-params", "testing", "wasm-runtime"]}
+namada_sdk = {path = "../sdk", default-features=false, features = ["download-params", "testing", "wasm-runtime"]}
 namada_test_utils = {path = "../test_utils"}
+namada_token = {path = "../token"}
```
Comment (tzemanovic) [inline crates/tests/Cargo.toml:33]:
```suggestion
```
use from sdk

File: crates/apps_lib/Cargo.toml (hunk 5 above)
Comment (batconjurer) [reply crates/apps_lib/Cargo.toml:34]: done

File: crates/light_sdk/Cargo.toml [hunk 10]
```diff
@@ -19,6 +19,7 @@ blocking = ["tokio"]
 namada-eth-bridge = ["namada_sdk/namada-eth-bridge"]
 
 [dependencies]
+namada_io = { path = "../io" }
```
Comment (tzemanovic) [inline crates/light_sdk/Cargo.toml:22]: ah sry to be pain - there's one more we can skip I think

Comment (batconjurer) [reply crates/light_sdk/Cargo.toml]: it's fixed

**Review by tzemanovic — APPROVED** [review]

**Comment by tzemanovic** [comment]
@mergifyio requeue

**Pull request merged**